main/src/library/Foldable.flix

```diff
@@ -80,24 +80,24 @@ pub class Foldable[t : Type -> Type] {
     ///
     /// Returns `false` if `t` is empty.
     ///
-    pub def exists(f: a -> Bool, t: t[a]): Bool =
-        Foldable.foldLeft((acc, x) -> acc or f(x), false, t)
+    pub def exists(f: a -> Bool, t: t[a]): Bool & ef =
+        Foldable.foldRightWithCont((x, k) -> k() or f(x), false, t)
 
     ///
     /// Returns `true` if and only if all elements in `t` satisfy the predicate `f`.
     ///
     /// Returns `true` if `t` is empty.
     ///
     pub def forall(f: a -> Bool, t: t[a]): Bool =
-        Foldable.foldLeft((acc, x) -> acc and f(x), true, t)
+        Foldable.foldRightWithCont((x, k) -> k() and f(x), true, t)
 
     ///
     /// Optionally returns the first element of `t` that satisfies the predicate `f` when searching from left to right.
     ///
     /// Returns `None` if `t` is empty.
     ///
     pub def findLeft(f: a -> Bool, t: t[a]): Option[a] =
-        Foldable.foldLeft((acc, x) -> Option.withDefault(default = if (f(x)) Some(x) else None, acc), None, t)
+        Foldable.foldRightWithCont((x, k) -> Option.withDefault(default = if (f(x)) Some(x) else None, k()), None, t)
```

Comment:
I think it should be possible to only call `k` when required.

Reply:
I'm not sure how to do that. I'm not even sure, that we can do `findLeft` with `foldRightWithCont`, since we get `f(x1, f(x2, ... f(xn, z)))`, and we want to find the first one from left to satisfy `f`

Reply:
The trick is this: When I call `k` determine when I do the right recursion. For findLeft, is just check `x` first. If thats the element I want then I am done and I dont have to call `k`. Thus i dont visit the rest of the list/foldable.

Reply:
Ahh, makes sense :)